### PR TITLE
MGMT-9720: Rewrite network configuration guide

### PR DESCRIPTION
**Pull request by mkowalski**
## List all the issues related to this PR

- [ ] New Feature 
- [ ] Enhancement 
- [ ] Bug fix
- [ ] Tests
- [x] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @
/cc @

## Checklist

- [ ] Title and description added to both, commit and PR.
- [ ] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [ ] Reviewers have been listed
- [ ] This change does not require a documentation update (docstring, `docs`, README, etc)
- [ ] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- Are the title and description (in both PR and commit) meaningful and clear?
- Is there a bug required (and linked) for this change?
- Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

### CODE REVIEW

File: docs/user-guide/network-configuration/cluster-networking.md [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+| serviceNetwork |                                       | IP address pool for services                                                                                                                                   |
+| machineNetwork |                                       | IP address blocks for machines forming the cluster                                                                                                             |
+| apiVIP         | api.<clustername.clusterdomain>       | The VIP to use for API communication. This setting must either be provided or pre-configured in the DNS so that the default name resolves correctly.  |
+| ingressVIP     | test.apps.<clustername.clusterdomain> | The VIP to use for ingress traffic                                                                                                                             |
+
+Apart from this, depending on the desired network stack, different network controller can be selected. Currently Assisted Service can deploy OCP clusters using one of the following configurations
```
Comment (avishayt) [inline docs/user-guide/network-configuration/cluster-networking.md:13]:
```suggestion
Apart from this, depending on the desired network stack, different network controllers can be selected. Currently, Assisted Service can deploy OCP clusters using one of the following configurations
```

File: docs/user-guide/network-configuration/cluster-networking.md [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+  clusterNetwork:
+  - cidr: 10.128.0.0/14
+    hostPrefix: 23
+```
+
+Creating a 3-node cluster using the snipet above may create the following network topology
+
+* pods scheduled in the node #1 get IPs from `10.128.0.0/23`
+* pods scheduled in the node #2 get IPs from `10.128.2.0/23`
+* pods scheduled in the node #3 get IPs from `10.128.4.0/23`
```
Comment (avishayt) [inline docs/user-guide/network-configuration/cluster-networking.md:54]:
```suggestion
* pods scheduled in node #1 get IPs from `10.128.0.0/23`
* pods scheduled in node #2 get IPs from `10.128.2.0/23`
* pods scheduled in node #3 get IPs from `10.128.4.0/23`
```

File: docs/user-guide/network-configuration/cluster-networking.md [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+
+Explaining OVN-K8s internals is out of scope of this document, but the pattern described above gives us an easy way to route pod-to-pod traffic between different nodes without keeping a big list of mapping between pods and their corresponding nodes.
+
+## Machine Network
+
+Machine network is a network used by all the hosts forming the cluster to communicate with each other. This is also the subnet that has to include API and Ingress VIPs.
```
Comment (avishayt) [inline docs/user-guide/network-configuration/cluster-networking.md:60]:
```suggestion
Machine network is a network used by all the hosts forming the cluster to communicate with each other. This is also the subnet that has to include the API and Ingress VIPs.
```

File: docs/user-guide/network-configuration/cluster-networking.md [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+## SNO vs Multi-Node Cluster
+
+Depending on whether a Single Node OpenShift or a Multi-Node cluster is deployed, different values are mandatory. The table below explains this in more detail.
+
+| Parameter      | SNO       | Multi-Node Cluster     |
+|----------------|-----------|------------------------|
+| clusterNetwork | Required  | Required               |
+| serviceNetwork | Required  | Required               |
+| machineNetwork | Required  | Forbidden in DHCP mode |
```
Comment (avishayt) [inline docs/user-guide/network-configuration/cluster-networking.md:70]: Link to dhcp-vip-allocation.md

File: docs/user-guide/network-configuration/cluster-networking.md [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+| apiVIP         | Forbidden | Required               |
+| ingressVIP     | Forbidden | Required               |
+
+## Airgapped environments
+
+The workflow for deploying a cluster without Internet access has some prerequisites which are out of scope of this document. You may consult the [Zero Touch Provisioning in the hard way git repo](https://github.com/jparrill/ztp-the-hard-way/blob/main/docs/disconnected-ZTP-flow-hub-deployment.md) for some insights.
```
Comment (avishayt) [inline docs/user-guide/network-configuration/cluster-networking.md:76]:
```suggestion
The workflow for deploying a cluster without Internet access has some prerequisites which are out of scope of this document. You may consult the [Zero Touch Provisioning the hard way git repo](https://github.com/jparrill/ztp-the-hard-way/blob/main/docs/disconnected-ZTP-flow-hub-deployment.md) for some insights.
```

File: docs/user-guide/network-configuration/dhcp-vip-allocation.md [hunk 6]
```diff
@@ -0,0 +1,19 @@
+# DHCP VIP Allocation
```
Comment (avishayt) [inline docs/user-guide/network-configuration/dhcp-vip-allocation.md:1]: Let's add a couple sentences as an intro:
1. What is it?
2. That it's not supported by OpenShift

File: docs/user-guide/network-configuration/dual-stack.md [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+
+* [Deploying a SNO cluster](../../hive-integration/crds/agentClusterInstall-dualstack-SNO.yaml)
+* [Deploying a multi-node OCP cluster](../../hive-integration/crds/agentClusterInstall-dualstack.yaml)
+
+## Sample REST API payload
+
+Deploying a SNO cluster
+
+```json
+XXX
```
Comment (avishayt) [inline docs/user-guide/network-configuration/dual-stack.md:17]: Fill in

File: docs/user-guide/network-configuration/dual-stack.md [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+```
+
+Deploying a multi-node OCP cluster
+
+```json
+XXX
```
Comment (avishayt) [inline docs/user-guide/network-configuration/dual-stack.md:23]: Fill in

File: docs/user-guide/network-configuration/single-stack.md [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+
+```json
+XXX
+```
+
+Deploying a multi-node OCP cluster
+
+```json
+XXX
+```
```
Comment (avishayt) [inline docs/user-guide/network-configuration/single-stack.md:47]: Fill in

File: docs/user-guide/network-configuration/static-configuration.md [hunk 10]
```diff
@@ -0,0 +1,15 @@
...
+
+## Sample KubeAPI CR
+
+[This sample CR](../../hive-integration/crds/nmstate.yaml) shows how to create a custom NMStateConfig to be used with Assisted Service on-premises.
+
+## Sample REST API payload
+
+```json
+XXX
+```
```
Comment (avishayt) [inline docs/user-guide/network-configuration/static-configuration.md:15]: Fill in

**Review by avishayt — COMMENTED** [review]
Really great work!  Thanks!

File: docs/user-guide/network-configuration/cluster-networking.md (hunk 5 above)
Comment (mkowalski) [reply docs/user-guide/network-configuration/cluster-networking.md:76]: Fixed, but it's the name in the upstream repo, just copy-paste

**Pull request merged**